app/controllers/support/devices/order_status_controller.rb

```diff
@@ -9,18 +9,16 @@ def update
     @form = Support::EnableOrdersForm.new(enable_orders_form_params)
     if @form.valid?
       ActiveRecord::Base.transaction do
```

Comment:
Should the transaction be sitting outside the service or inside?

Reply:
That's a really good question, and I thought about it a lot while coding this - but I'm pretty sure it should be outside the service. 

The service doesn't have any knowledge of the context in which it's called, only the caller does - in this case the controller, but it could also be called from some other larger process like a rake task.  The right thing to do is to let the exception bubble out and the appropriate remedial action be taken by the invoking context. Related, on some previous apps I've done things like enforce transactions for all non-GET requests by adding an `around_filter` in ApplicationController

The only case for putting it inside the service I can see would be if the service could be called remotely (like a REST API) - but then that's effectively at controller level anyway.